connector/signaltometricsconnector/config/config.go

```diff
@@ -299,6 +319,36 @@ func validateMetricInfo[K any](mi MetricInfo, parser ottl.Parser[K]) error {
 			return fmt.Errorf("failed to parse value OTTL expression for summary: %w", err)
 		}
 	}
+	if mi.Gauge != nil {
+		metricsDefinedCount++
+		if _, err := parser.ParseValueExpression(mi.Gauge.Value); err != nil {
+			return fmt.Errorf("failed to parse value OTTL expression for gauge: %w", err)
+		}
+		// Custom validation: fail if ExtractGrokPatterns is used with multiple patterns or configured with unsupported type
+		if strings.Contains(mi.Gauge.Value, "ExtractGrokPatterns") {
```

Comment:
I am thinking to limit the support for defined OTTL expressions to return only a numerical value - anything else could be rejected during runtime with an error log (see the above comment for more details).